README.md

```diff
@@ -1,10 +1,157 @@
+
+---
+
+## Overview
+
+This rock builds **Apache Airflow** from source on **Ubuntu** and packages it as a Pebble-managed OCI container.
+It includes a wide range of official Airflow providers and runs Airflow in standalone mode by default.
+
+**Key features:**
+
+* Builds Airflow directly from the upstream GitHub tag (e.g. `3.1.0`).
+* Uses the official constraints file for Python to ensure dependency compatibility.
+* Stages common Airflow provider packages.
+* Runs Airflow in standalone mode automatically via Pebble upon startup
+* Includes licensing information for both Airflow and this rock.
+
+
+## Project Structure
+
+```
+airflow-rocks/
+├─ 3.1.0/
+│  ├─ rockcraft.yaml         # Rockcraft manifest defining the rock
+│  ├─ goss.yaml
+│  ├─ goss_wait.yaml
+├─ DEVELOPING.md          
+├─ README.md
+├─ justfile
+```
+
+## Building the rock
+
+To build the rock:
+
+```bash
+just pack ${version}
+```
+
+This will:
+
+* Fetch Airflow source from GitHub (`3.1.0` tag)
+* Build Airflow and providers into a Python environment
+* Package it as an OCI image wrapped in a `.rock` file (e.g., `airflow-rock_3.1.0_amd64.rock`)
+
+## Running and testing the rock
+
+Once built, you can run the rock locally using `docker`:
+
+### Using Docker
+
+```bash
+# Load the rock into your local Docker daemon
+rockcraft.skopeo --insecure-policy copy oci-archive:airflow-rock_3.1.0_amd64.rock docker-daemon:airflow-rock:3.1.0
+# Run the rock
+docker run -it --rm -p 5000:5000 airflow-rock:3.1.0
+```
+
+The container will start Airflow in standalone mode.
+The Airflow web UI will be available at: http://localhost:5000
+
+### Using Pebble inside the container
+
+```bash
+docker exec -it <container_id> pebble services
+docker exec -it <container_id> pebble logs
+```
+
+## Rock services and checks
+
+The rock defines one service, managed by Pebble:
+
+```yaml
+services:
+  airflow:
+    startup: enabled
+    command: /usr/bin/airflow standalone
+```
+
+This runs Airflow’s built-in standalone mode, which launches:
+
+* An API-Server
+* A scheduler
+* A metadata database (SQLite, by default)
+
+A simple liveness check verifies Airflow is running:
+
+```yaml
+checks:
+  airflow-running:
+    exec:
+      command: pgrep airflow
+```
+
+## Parts Explained
+
+The rock is built from several parts:
+
+| Part                       | Purpose                                                                 |
+| -------------------------- | ----------------------------------------------------------------------- |
+| **airflow**                | Builds Apache Airflow from source and installs providers                |
+| **airflow-license**        | Includes upstream Apache Airflow LICENSE                                |
+| **airflow-rock-license**   | Includes the LICENSE for this rock                                      |
+| **env_stage** *(optional)* | Stages a system-wide `/etc/environment` file with environment variables |
```

Comment:
please confirm if the `env_stage` part still exists - i was under the assumption that it was removed